### PR TITLE
nixos/jellyseerr: add package option

### PR DESCRIPTION
**Pull request by sielicki**
## Description of changes

jellyseerr (due to nextjs limitations) cannot be reverse-proxied under a sub-path unless it's rebuilt.

Rebuilding it to work under a sub-path is trivial (although not implemented in this PR and/or exposed in nixpkgs), but upon doing so, this service will still hardcode the original jellyseerr package unless an overlay is used.

Expose this as an option to make it easier.

### CODE REVIEW

**Review by getchoo — APPROVED** [review]
lgtm

edit: this will need to be rebased, though. oops!

**Comment by J-Swift** [comment]
@sielicki found this when looking into doing the same. Thanks for pushing the PR! Any chance you can point to an example of successfully using the `basePath` on the custom package? When I try, I get redirect loops...

File: nixos/modules/services/misc/jellyseerr.nix [hunk 1]
```diff
@@ -32,10 +33,10 @@ in
       serviceConfig = {
         Type = "exec";
         StateDirectory = "jellyseerr";
-        WorkingDirectory = "${pkgs.jellyseerr}/libexec/jellyseerr/deps/jellyseerr";
+        WorkingDirectory = "${cfg.package}/libexec/jellyseerr/deps/jellyseerr";
         DynamicUser = true;
-        ExecStart = "${pkgs.jellyseerr}/bin/jellyseerr";
-        BindPaths = [ "/var/lib/jellyseerr/:${pkgs.jellyseerr}/libexec/jellyseerr/deps/jellyseerr/config/" ];
+        ExecStart = "${cfg.package}/bin/jellyseerr";
```
Comment (getchoo) [inline nixos/modules/services/misc/jellyseerr.nix:38]:
```suggestion
        ExecStart = lib.getExe cfg.package;
```

**Review by getchoo — CHANGES_REQUESTED** [review]

**Comment by eyJhb** [comment]
@sielicki I'm also curious on how you override the path, because if I just do

```nix
    package = pkgs.jellyseerr.overrideAttrs (old: {
      postPatch = ''
        sed -i "5i basePath: '/jellyseerr'," next.config.js
      '';
    });
```

Then it still redirects me to `/login`.

**Pull request merged**